### PR TITLE
Initial examples

### PR DESCRIPTION
**Pull request by chessbyte**

### CODE REVIEW

**Comment by chessbyte** [comment]
@Fryguy Please review

File: examples/client_example.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+rv = netapp_client.volume_list_info do
+  volume "vol1"
+end
+puts rv.inspect
+
+puts
+puts "*** List info for a specific volume V4:"
+rv = netapp_client.volume_list_info { |vla|
+  vla.volume = "vol1"
+}
```
Comment (Fryguy) [inline examples/client_example.rb]: Minor: do/end notation

File: examples/api_example.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "net_app_manageability"
+svr = NetAppManageability::API.server_open(SERVER, 1, 1)
+NetAppManageability::API.server_style(svr, NetAppManageability::API::NA_STYLE_LOGIN_PASSWORD)
+NetAppManageability::API.server_adminuser(svr, USERNAME, PASSWORD)
+rv = NetAppManageability::API.server_invoke(svr, "volume-list-info", :volume => "vol1")
+
+puts "RV: #{rv.inspect}"
```
Comment (Fryguy) [inline examples/api_example.rb]: Minor...line feed at end.

File: examples/create_volume.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+#
+# Add a list of root access hosts to the rules.
+# These are the ESX hosts that will be able to access the datastore.
+#
+rules = rv.rules
+rules.exports_rule_info.root = NmaHash.new do
+  exports_hostname_info NmaArray.new do
+    ROOT_HOSTS.each do |rh|
+      push NmaHash.new { name rh }
```
Comment (Fryguy) [inline examples/create_volume.rb]: o_O what a weird notation

File: examples/list_volumes.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  username   USERNAME
+  password   PASSWORD
+end
+
+rv = netapp_client.volume_list_info
+
+aia = rv.volumes.volume_info.to_ary
+aia.each do |ai|
+  puts "Volume: #{ai.name}\tavailable space: #{ai.size_available}"
+end
```
Comment (Fryguy) [inline examples/list_volumes.rb]: Minor...line feed at end

**Comment by Fryguy** [comment]
Looks great...everything I mentioned is minor.

File: examples/create_volume.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+# Get the export rules for the new volume's NFS share.
+#
+rv = netapp_client.nfs_exportfs_list_rules(:pathname, NFS_PATH)
+
+#
+# Add a list of root access hosts to the rules.
+# These are the ESX hosts that will be able to access the datastore.
+#
+rules = rv.rules
+rules.exports_rule_info.root = NmaHash.new do
```
Comment (Fryguy) [inline examples/create_volume.rb]: Oh this should be NAMHash or probably NetAppManageability::NAMHash

File: examples/nfs_root_hosts.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+netapp_client = NetAppManageability::Client.new do
+  server     SERVER
+  auth_style NetAppManageability::Client::NA_STYLE_LOGIN_PASSWORD
+  username   USERNAME
+  password   PASSWORD
+end
+
+rv = netapp_client.nfs_exportfs_list_rules(:pathname, "/vol/#{VOLUME_NAME}")
+raise "No export rules found for path /vol/#{VOLUME_NAME}" unless rv.kind_of?(NmaHash)
```
Comment (Fryguy) [inline examples/nfs_root_hosts.rb]: Same on NAMHash

File: examples/lun_info.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+rv = netapp_client.lun_map_list_info(:path, '/vol/luns/lun1')
+puts rv.inspect
+puts
+
+rv = netapp_client.iscsi_adapter_initiators_list_info
+puts rv.inspect
+puts
+
+rv = nnetapp_clientma.iscsi_connection_list_info
```
Comment (Fryguy) [inline examples/lun_info.rb]: Typo `nnetapp_clientma` -> `netapp_client`

**Pull request merged**